server/helpers/cache.js

```diff
@@ -0,0 +1,38 @@
+const TIMEOUT = 24 * 60 * 60 * 1000 // 1 day
+const PRUNE_RATIO = 0.1
+
+class Cache {
+  constructor(limit, maxAge = TIMEOUT) {
+    this.limit = limit
+    this.maxAge = maxAge
+    this.cache = {}
+    this.size = 0
+  }
+
+  get(key) {
+    if (key in this.cache && this.cache[key].timestamp + this.maxAge > Date.now()) {
```

Comment:
I'd consider deleting the stale keys right away and then you could split the concern of enforcing the record max age and accessing non-stale records of the cache

Reply:
Not sure if I understood this comment correctly. I added deletion of old entries right away.

Reply:
yes, that's what I meant :+1: